packages/grafana-data/src/transformations/transformers/joinDataFrames.ts

```diff
@@ -47,6 +47,11 @@ export interface JoinOptions {
    */
   joinBy?: FieldMatcher;
 
+  /**
+   * The fields to join on
+   */
+  fields?: { [key: string]: string };
```

Comment:
is this parallel to the "joinBy"?

Can the frame/field selection be part of a FieldMatcher (and config?)?  -- then we can use the logic/ui all the places that need to pick fields out of a `[]frame`

Reply:
Yes it is parallel to joinBy, which takes a FieldMatcher

Currently the transformer create FieldMatchers based on frame.refId and field.name. Do you suggest to allow passing FieldMatchers like for joinBy?

Reply:
think this needs a bit of rethink